src/ppl/src/IOPlacer.cpp

```diff
@@ -485,10 +484,10 @@ void IOPlacer::findSections(int begin, int end, Edge edge, std::vector<Section>&
   }
 }
 
-std::vector<Section> IOPlacer::createSectionsPerConstraint(const Constraint &constraint)
+void IOPlacer::createSectionsPerConstraint(Constraint &constraint)
 {
-  const Interval &interv = constraint.interval;
-  const Edge &edge = interv.edge;
+  Interval &interv = constraint.interval;
+  Edge &edge = interv.edge;
```

Comment:
Why make these non-const? Why change this method at all and not just handle the return value with

constraint.sections = createSectionsPerConstraint(constraint);